### PR TITLE
Force unsigned short for LineNumber.toString()

### PR DESCRIPTION
**Pull request by markro49**
Ensure startPc and lineNumber are treated as unsigned short in LineNumber.toString()

### CODE REVIEW

File: src/main/java/org/apache/bcel/classfile/LineNumber.java [hunk 1]
```diff
@@ -96,15 +96,15 @@ public void dump( final DataOutputStream file ) throws IOException {
      * @return Corresponding source line
      */
     public int getLineNumber() {
-        return 0xffff & lineNumber;
+        return lineNumber & 0xffff;
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/classfile/LineNumber.java:99]: Is this a style change? If not a comment is needed.

File: src/main/java/org/apache/bcel/classfile/LineNumber.java [hunk 2]
```diff
@@ -129,7 +129,7 @@ public void setStartPC( final int startPc ) {
      */
     @Override
     public String toString() {
-        return "LineNumber(" + startPc + ", " + lineNumber + ")";
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/classfile/LineNumber.java:132]: Can we call the getters here?

File: src/test/java/org/apache/bcel/data/LargeMethod.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.bcel.data;
+
+    // Due to the javac implemtation of try finally, this class
```
Comment (garydgregory) [inline src/test/java/org/apache/bcel/data/LargeMethod.java:21]: I think we need to mention a Java version as different compilers might generate different bytes codes, maybe?

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @markro49 
Please see my comments.

File: src/main/java/org/apache/bcel/classfile/LineNumber.java [hunk 4]
```diff
@@ -96,15 +96,15 @@ public void dump( final DataOutputStream file ) throws IOException {
...
+        return lineNumber & 0xffff;
     }
 
 
     /**
      * @return PC in code
      */
     public int getStartPC() {
-        return  0xffff & startPc;
+        return startPc & 0xffff;
```
[the remark this answers is not in the document]
Comment (markro49) [reply src/main/java/org/apache/bcel/classfile/LineNumber.java:107]: style change

File: src/main/java/org/apache/bcel/classfile/LineNumber.java (hunk 1 above)
Comment (markro49) [reply src/main/java/org/apache/bcel/classfile/LineNumber.java:99]: style change

File: src/main/java/org/apache/bcel/classfile/LineNumber.java (hunk 2 above)
Comment (markro49) [reply src/main/java/org/apache/bcel/classfile/LineNumber.java:132]: good idea - much cleaner code

File: src/test/java/org/apache/bcel/data/LargeMethod.java (hunk 3 above)
Comment (markro49) [reply src/test/java/org/apache/bcel/data/LargeMethod.java:21]: I'll expand the comment.

File: src/test/java/org/apache/bcel/PLSETestCase.java [hunk 5]
```diff
@@ -116,6 +119,25 @@ public void testB295() throws Exception
         assertEquals(lv.getLength(), new_lv.getLength(), "live range length");
     }
 
+    /**
+     * BCEL-361: LineNumber.toString() treats code offset as signed
+     */
+    @Test
+    public void testB361() throws Exception
+    {
+        final JavaClass clazz = getTestClass(PACKAGE_BASE_NAME+".data.LargeMethod");
```
Comment (garydgregory) [inline src/test/java/org/apache/bcel/PLSETestCase.java:128]: Missing spaces around `+`. I'm on vacation but there is a test I want to run... back in a while...

**Comment by garydgregory** [comment]
Hi @markro49 
Do you have any thoughts after storing the shorts with their "proper" unsigned values in the instance variables in the first place?

**Pull request merged**